sign.go

```diff
@@ -29,3 +52,68 @@ func SignInTotoStatement(ctx context.Context, stmt intoto.Statement, provider cl
 
 	return env, nil
 }
+
+func SignInTotoStatementExt(ctx context.Context, stmt intoto.Statement, provider client.OpenIdProvider) (*Envelope, error) {
+	tl, ok := ctx.Value(tlCtxKey(DefaultCtxKey)).(TL)
+	if !ok {
+		tl = &RekorTL{}
+	}
+
+	// encode in-toto statement
+	payload, err := json.Marshal(stmt)
+	if err != nil {
+		return nil, err
+	}
+	env := new(Envelope)
+	env.Payload = base64.StdEncoding.EncodeToString(payload)
+	env.PayloadType = intoto.PayloadType
+	encPayload := dsse.PAE(intoto.PayloadType, payload)
+
+	// statement message digest
+	hash := s256(encPayload)
+	hashHex := hex.EncodeToString(hash)
+
+	// generate ephemeral keys to sign message digest
+	signer, err := util.GenKeyPair(jwa.ES256)
+	if err != nil {
+		return nil, fmt.Errorf("error generating key pair: %w", err)
+	}
+	sig, err := signer.Sign(rand.Reader, hash, crypto.SHA256)
+	if err != nil {
+		return nil, err
+	}
+	ecPub := signer.Public().(*ecdsa.PublicKey)
```

Comment:
Is this cast guaranteed to work?

Reply:
no, good catch! [4af6ebb](https://github.com/openpubkey/signed-attestation/pull/4/commits/4af6ebb12dcb8ad9b9c9899585d96cf2c50c9451)